lib/auth/webauthn/register.go

```diff
@@ -48,6 +50,57 @@ type RegistrationIdentity interface {
 	DeleteWebauthnSessionData(ctx context.Context, user, sessionID string) error
 }
 
+// WithInMemorySessionData returns a RegistrationIdentity implementation that
+// keeps SessionData in memory.
+func WithInMemorySessionData(identity RegistrationIdentity) RegistrationIdentity {
+	return &inMemoryIdentity{
+		RegistrationIdentity: identity,
+		sessionData:          make(map[string]*wantypes.SessionData),
+	}
+}
+
+type inMemoryIdentity struct {
+	RegistrationIdentity
+
+	// mu guards the fields below it.
+	// We don't foresee concurrent use for inMemoryIdentity, but it's easy enough
+	// to play it safe.
+	mu          sync.RWMutex
+	sessionData map[string]*wantypes.SessionData
+}
+
+func (identity *inMemoryIdentity) UpsertWebauthnSessionData(ctx context.Context, user, sessionID string, sd *wantypes.SessionData) error {
+	identity.mu.Lock()
+	defer identity.mu.Unlock()
+	identity.sessionData[sessionDataKey(user, sessionID)] = sd
+	return nil
+}
+
+func (identity *inMemoryIdentity) GetWebauthnSessionData(ctx context.Context, user, sessionID string) (*wantypes.SessionData, error) {
+	identity.mu.RLock()
+	defer identity.mu.RUnlock()
+	sd, ok := identity.sessionData[sessionDataKey(user, sessionID)]
```

Comment:
What's the caller going to do with `sd`? Any concerns about concurrent modifications?

Reply:
The caller is the registration package itself, it just reads it. Left a comment.